src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/database/jpa/mapper/induction/FutureWorkInterestsEntityMapper.kt

```diff
@@ -29,21 +31,18 @@ abstract class FutureWorkInterestsEntityMapper {
   @Autowired
   private lateinit var entityListManager: InductionEntityListManager<WorkInterestEntity, WorkInterest>
 
+  @BeanMapping(qualifiedByName = ["addNewInterestsDuringCreate"])
   @ExcludeJpaManagedFieldsIncludingDisplayNameFields
   @GenerateNewReference
   @Mapping(target = "createdAtPrison", source = "prisonId")
   @Mapping(target = "updatedAtPrison", source = "prisonId")
   @Mapping(target = "interests", ignore = true)
   abstract fun fromCreateDtoToEntity(dto: CreateFutureWorkInterestsDto): FutureWorkInterestsEntity
 
+  @Named("addNewInterestsDuringCreate")
   @AfterMapping
-  fun addInterests(dto: CreateFutureWorkInterestsDto, @MappingTarget entity: FutureWorkInterestsEntity) {
-    dto.interests.forEach {
-      entity.addChild(
-        workInterestEntityMapper.fromDomainToEntity(it),
-        entity.interests(),
-      )
-    }
+  fun addNewInterestsDuringCreate(dto: CreateFutureWorkInterestsDto, @MappingTarget entity: FutureWorkInterestsEntity) {
+    addNewInterests(dto.interests, entity)
```

Comment:
A simple shared method to add new interests, either when creating an Induction (i.e. via `CreateFutureWorkInterestsDto`) or when updating it (via `UpdateFutureWorkInterestsDto`).